examples/flutter_gallery/lib/demo/pesto_demo.dart

```diff
@@ -0,0 +1,683 @@
+          ]
+        )
+      )
+    );
+  }
+
+  Widget _buildCard(BuildContext context, Recipe recipe) {
+    final TextStyle titleStyle = _textStyle(24.0, FontWeight.w600);
+    final TextStyle authorStyle = _textStyle(12.0, FontWeight.w500).copyWith(color: Colors.black54);
+
+    return new Padding(
+      padding: const EdgeInsets.all(4.0),
+      child: new GestureDetector(
+        onTap: () => _showRecipe(context, recipe),
+        child: new Card(
+          child: new Column(
+            crossAxisAlignment: CrossAxisAlignment.start,
+            children: <Widget>[
+              new AssetImage(
+                name: recipe.imagePath,
+                fit: ImageFit.scaleDown
+              ),
+              new Flexible(child:
+                new Row(
+                  children: <Widget>[
+                    new Padding(
+                      padding: const EdgeInsets.all(16.0),
+                      child: new AssetImage(
+                        width: 48.0,
+                        height: 48.0,
+                        name: recipe.ingredientsImagePath
+                      )
+                    ),
+                    new Column(
+                      crossAxisAlignment: CrossAxisAlignment.start,
+                      mainAxisAlignment: MainAxisAlignment.center,
+                      children: <Widget>[
+                        new Text(recipe.name, style: titleStyle),
+                        new Text(recipe.author, style: authorStyle),
+                      ]
+                    )
+                  ]
+                )
+              )
+            ]
+          )
+        )
+      )
+    );
+  }
+
+  void _showFavorites(BuildContext context) {
+    Navigator.push(context, new MaterialPageRoute<Null>(
+      builder: (BuildContext context) {
+        return new PestoDemo(showFavorites: true);
+      }
+    ));
+  }
+
+  void _showRecipe(BuildContext context, Recipe recipe) {
+    Navigator.push(context, new MaterialPageRoute<Null>(
+      builder: (BuildContext context) {
+        return new Theme(
+          data: _kTheme,
+          child: new _RecipePage(recipe: recipe)
+        );
+      }
+    ));
+  }
+}
+
+class _RecipePage extends StatefulWidget {
+  _RecipePage({ Key key, this.recipe }) : super(key: key);
+
+  final Recipe recipe;
+
+  @override
+  _RecipePageState createState() => new _RecipePageState();
+}
+
+class _RecipePageState extends State<_RecipePage> {
+  @override
+  Widget build(BuildContext context) {
+    ThemeData theme = Theme.of(context);
+    TextStyle titleStyle = _textStyle(34.0);
+    TextStyle descriptionStyle = _textStyle(15.0).copyWith(color: Colors.black54, height: 24.0/15.0);
+    TextStyle itemStyle = _textStyle(15.0).copyWith(height: 24.0/15.0);
+    TextStyle itemAmountStyle = itemStyle.copyWith(color: theme.primaryColor);
+    TextStyle headingStyle = itemStyle.copyWith(fontSize: 16.0, fontWeight: FontWeight.bold);
+
+    TableRow _buildItemRow(String left, String right) {
```

Comment:
Per the comment about "big" above.  Maybe a RecipeStep widget instead of this function?